frontend/test/metabase/scenarios/admin/datamodel/metadata.cy.spec.js

```diff
@@ -33,4 +33,44 @@ describe("scenarios > admin > datamodel > metadata", () => {
     openOrdersTable();
     cy.findAllByText("Awesome Concrete Shoes");
   });
+
+  it("should correctly apply and display custom remapping for numeric values", () => {
+    const customMap = {
+      1: "Awful",
+      2: "Unpleasant",
+      3: "Meh",
+      4: "Enjoyable",
+      5: "Perfecto",
+    };
+
+    // go directly to Data Model page for Sample Dataset
+    cy.visit("/admin/datamodel/database/1");
+    // edit "Rating" values in "Reviews" table
+    cy.findByText("Reviews").click();
+    cy.findByDisplayValue("Rating")
+      .parent()
+      .find(".Icon-gear")
+      .click();
+
+    // apply custom remapping for "Rating" values 1-5
+    cy.findByText("Use original value").click();
+    cy.findByText("Custom mapping").click();
+    cy.findByText(
+      "You might want to update the field name to make sure it still makes sense based on your remapping choices.",
+    );
+
+    Object.keys(customMap).forEach(key => {
```

Comment:
Not important, but in case you haven't seen it, you can use `Object.entries` to iterate over `[key, value]` pairs.

Reply:
Great to know! Thanks :)

Reply:
I know you said it's not important, but since I already had to commit (delete older test) I updated this to use `Object.entries`. It's best to learn/remember by direct application.